opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/SecretKeyServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package de.adorsys.opba.protocol.facade.services;
+
+import com.google.crypto.tink.Aead;
+import de.adorsys.opba.protocol.api.services.SecretKeyService;
+import de.adorsys.opba.protocol.facade.config.EncryptionProperties;
+import de.adorsys.opba.protocol.facade.config.FacadeSecurityProvider;
+import lombok.RequiredArgsConstructor;
+import lombok.SneakyThrows;
+import org.springframework.stereotype.Service;
+
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import java.util.Base64;
+
+import static de.adorsys.opba.protocol.facade.utils.EncryptionUtils.getNewSalt;
+
+@Service
+@RequiredArgsConstructor
+public class SecretKeyServiceImpl implements SecretKeyService {
```

Comment:
probably SecretKeyOperations is better name. Service is typically one-purpose java class

Reply:
renamed